packages/common/src/hardhat/tasks/compile.js

```diff
@@ -29,13 +25,9 @@ internalTask(TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS, async (_, { config, network
   // Build absolute path for all directories on user-specified whitelist.
   const whitelist = config.networks[network.name].compileWhitelist;
   if (whitelist && Array.isArray(whitelist)) {
-    const whitelistDirs = whitelist.map((x) => {
-      return path.resolve(CONTRACTS_DIR, x);
-    });
-
     filePaths = filePaths.filter((filePath) => {
-      for (let whitelistedDir of whitelistDirs) {
```

Comment:
OOC, are we back to square one on this file? haha

Reply:
it's a fair amount simpler than the original I think.